Improve error cases

pierrewillenbrockdfki opened the pull request:
This adds some more error messages and changes one assert to a message+throw because the assertion is not backed by documentation or code structure, so aborting the process seems a bit harsh.

pierrewillenbrockdfki committed:
Change one assert to test+throw

The caller is not limited in its choice of timestamps, and if it
is not using the aggregator for all ports, it may actually end up
asking for a future timestamp, like it could for a past timestamp.
Instead of aborting the app, output an error message and throw an
exception.

doudou commented:
I don't mind _per se_ ... but can you explain the need ? Anything that catches exceptions generically should *really* output the message already.

pierrewillenbrockdfki commented:
> I don't mind _per se_ ... but can you explain the need ? Anything that catches exceptions generically should _really_ output the message already.

The messages are more for the benefit of the person that has to deal with tasks that don't catch the exception. Not sure if orogen/rtt does anything about exceptions leaking from the hooks, though. If that were the case, i would wonder why the messages seem to not make it to the log file.

doudou commented:
I am wondering exactly that. The message of exceptions that are caught "leaking" off the hooks are displayed

https://github.com//rock-core/rtt/blob/baaea5022bc79f94770a962e2e16cc0cb28cde0b/rtt/ExecutionEngine.cpp#L399

doudou commented:
Maybe the transformer/aggregator isn't ... in which case the best would be to fix that.

pierrewillenbrockdfki commented:
I see the discussion in https://github.com/rock-core/drivers-transformer/pull/20. I'll try to figure out what path is failing to log, maybe drop the extra logging part.

pierrewillenbrockdfki commented:
My current theory is that for all deployments, RTT ends up logging to the same orocos.log file, while base::logging is always logging to stderr, which is redirected at a separate file. This would be entirely our(dfki) own fault for not setting ORO_LOGFILE to a sensible value in all cases.

doudou commented:
The what() is output also to stdout/stderr. At least, these error messages do end up in the "normal" log file.

pierrewillenbrockdfki commented:
True, but only if ORO_LOGLEVEL >= 3(Error); we run at ORO_LOGLEVEL=2(Critical); the file gets everything up to 5(Info)(Edit, or higher if ORO_LOGLEVEL is higher).

pierrewillenbrockdfki commented:
Dropped the changes to the `NonAligningTransfomer`. Once we have fixed our process startup no extra messages are needed. Kept the one where the assertion was replaced for consistency.
